src/Sdk.php

```diff
@@ -8,11 +8,13 @@
 /**
  * Creates a new Client and Hub which will be set as current.
  *
- * @param array $options
+ * @param array              $options       the Client options
+ * @param null|ClientBuilder $clientBuilder Use this instance of ClientBuilder instead of creating a new one
  */
-function init(array $options = []): void
+function init(array $options = [], ?ClientBuilder $clientBuilder = null): void
 {
-    Hub::setCurrent(new Hub(ClientBuilder::create($options)->getClient()));
+    $builder = $clientBuilder ?? ClientBuilder::create($options);
+    Hub::setCurrent(new Hub($builder->getClient()));
```

Comment:
Please add a blank line above this one to improve readability. One thing that puzzles me is that accepting an instance of the client builder means that it has to be created using certain options but the `init` method accept the options too. This in turn means that the first argument is useless if you're using the second argument and this may not be a clear enough API for the user... Just wondering though, we may leave it there and just postpone the problem to a later PR

Reply:
Created a new issue for this: https://github.com/getsentry/sentry-php/issues/695